integrations/chroma/src/haystack_integrations/document_stores/chroma/document_store.py

```diff
@@ -363,50 +350,47 @@ def _normalize_filters(filters: Dict[str, Any]) -> Tuple[List[str], Dict[str, An
         passed to `ids`, `where` and `where_document` respectively.
         """
         if not isinstance(filters, dict):
-            msg = "'filters' parameter must be a dictionary"
+            msg = f"'{filters}' parameter must be a dictionary"
             raise ChromaDocumentStoreFilterError(msg)
 
         ids = []
         where = defaultdict(list)
-        where_document = defaultdict(list)
-        keys_to_remove = []
+        where_document = {}
+        document_flag = False
 
-        for field, value in filters.items():
+        # Process filters
+        for field, value in list(filters.items()):
             if field == "content":
-                # Schedule for removal the original key, we're going to change it
-                keys_to_remove.append(field)
                 where_document["$contains"] = value
+                document_flag = True
+                del filters[field]
             elif field == "id":
-                # Schedule for removal the original key, we're going to change it
-                keys_to_remove.append(field)
                 ids.append(value)
-            elif isinstance(value, (list, tuple)):
-                # Schedule for removal the original key, we're going to change it
-                keys_to_remove.append(field)
-
-                # if the list is empty the filter is invalid, let's just remove it
-                if len(value) == 0:
+                del filters[field]
+            elif isinstance(value, (list, tuple)) and field not in ["$and", "$or"]:
```

Comment:
You are right. I think its a mistake, because the chromadb filters also don't support tuples. So we basically don't have this functionality. There is no mention of them in docs so we can just remove them from here?

Reply:
Ah, then yeah - We can remove it without a deprecation (counts as a bug).